3.2_interlude.ipynb

```diff
@@ -84,55 +85,54 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "Driver(() => new Module {\n",
+    "test(new Module {\n",
     "    // Example circuit using a Queue\n",
     "    val io = IO(new Bundle {\n",
     "      val in = Flipped(Decoupled(UInt(8.W)))\n",
     "      val out = Decoupled(UInt(8.W))\n",
     "    })\n",
     "    val queue = Queue(io.in, 2)  // 2-element queue\n",
     "    io.out <> queue\n",
-    "  }) { c => new PeekPokeTester(c) {\n",
-    "    // Example testsequence showing the use and behavior of Queue\n",
-    "    poke(c.io.out.ready, 0)\n",
-    "    poke(c.io.in.valid, 1)  // Enqueue an element\n",
-    "    poke(c.io.in.bits, 42)\n",
+    "  }) { c =>\n",
+    "    c.io.out.ready.poke(false.B)\n",
+    "    c.io.in.valid.poke(true.B)  // Enqueue an elemen.Ut\n",
+    "    c.io.in.bits.poke(42.U)\n",
     "    println(s\"Starting:\")\n",
-    "    println(s\"\\tio.in: ready=${peek(c.io.in.ready)}\")\n",
-    "    println(s\"\\tio.out: valid=${peek(c.io.out.valid)}, bits=${peek(c.io.out.bits)}\")\n",
-    "    step(1)\n",
-    "  \n",
-    "    poke(c.io.in.valid, 1)  // Enqueue another element\n",
-    "    poke(c.io.in.bits, 43)\n",
+    "    println(s\"\\tio.in: ready=${c.io.in.ready.peek().litValue}\")\n",
```

Comment:
What about just printing the output of peek (without the litValue extraction), which also includes structural data (eg, uint width)?

Reply:
Done, actually I did it both ways and added a note describing the difference